From 48ea1c467ca2002de244cbca87134fd8e2084eda Mon Sep 17 00:00:00 2001
From: Alexander Mikhaylenko <alexm@gnome.org>
Date: Wed, 30 Sep 2020 17:22:42 +0500
Subject: [PATCH] [GTK] Chassis type check fails if the value is quoted
 https://bugs.webkit.org/show_bug.cgi?id=217123

Reviewed by NOBODY (OOPS!).

Sometimes the chassis type can be quoted, in this case we'll fail to
parse it and fall back to desktop type. Unquote the string before
attempting to parse it.

* wtf/glib/ChassisType.cpp:
(WTF::readMachineInfoChassisType):
---
 Source/WTF/wtf/glib/ChassisType.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Source/WTF/wtf/glib/ChassisType.cpp b/Source/WTF/wtf/glib/ChassisType.cpp
index 3fc326e3ba..b8269de3b7 100644
--- a/Source/WTF/wtf/glib/ChassisType.cpp
+++ b/Source/WTF/wtf/glib/ChassisType.cpp
@@ -46,7 +46,12 @@ static Optional<ChassisType> readMachineInfoChassisType()
     for (int i = 0; split.get()[i]; ++i) {
         if (g_str_has_prefix(split.get()[i], "CHASSIS=")) {
             char* chassis = split.get()[i] + 8;
-            if (!strcmp(chassis, "tablet") || !strcmp(chassis, "handset"))
+
+            GUniquePtr<char> unquoted(g_shell_unquote(chassis, &error.outPtr()));
+            if (error.get() != nullptr)
+                g_warning("Could not unquote chassis type %s: %s", chassis, error->message);
+
+            if (!strcmp(unquoted.get(), "tablet") || !strcmp(unquoted.get(), "handset"))
                 return ChassisType::Mobile;
 
             return ChassisType::Desktop;
-- 
2.26.2

